fog/ledger/server/src/router_handlers.rs

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use crate::error::{router_server_err_to_rpc_status, RouterServerError};
+use futures::{future::try_join_all, SinkExt, TryStreamExt};
+use grpcio::{ChannelBuilder, DuplexSink, RequestStream, RpcStatus, WriteFlags};
+use mc_attest_api::attest;
+use mc_attest_enclave_api::{EnclaveMessage, NonceSession};
+use mc_common::{logger::Logger, ResponderId};
+use mc_fog_api::{
+    ledger::{
+        LedgerRequest, LedgerResponse, MultiKeyImageStoreRequest, MultiKeyImageStoreResponse,
+        MultiKeyImageStoreResponseStatus,
+    },
+    ledger_grpc::KeyImageStoreApiClient,
+};
+use mc_fog_ledger_enclave::LedgerEnclaveProxy;
+use mc_fog_uri::{ConnectionUri, KeyImageStoreUri};
+use mc_util_grpc::{rpc_invalid_arg_error, ConnectionUriGrpcioChannel};
+use std::{collections::BTreeMap, str::FromStr, sync::Arc};
+
+const RETRY_COUNT: usize = 3;
```

Comment:
I'm thinking the retry count should be a config option that defaults to 3 rather than a constant of 3.

Reply:
You're probably right. Will do.